core/src/main/java/org/apache/spark/shuffle/sort/ShuffleExternalSorter.java

```diff
@@ -108,42 +102,20 @@ public ShuffleExternalSorter(
       int initialSize,
       int numPartitions,
       SparkConf conf,
-      ShuffleWriteMetrics writeMetrics) throws IOException {
+      ShuffleWriteMetrics writeMetrics) {
+    super(memoryManager);
     this.taskMemoryManager = memoryManager;
     this.blockManager = blockManager;
     this.taskContext = taskContext;
-    this.initialSize = initialSize;
     this.peakMemoryUsedBytes = initialSize;
     this.numPartitions = numPartitions;
     // Use getSizeAsKb (not bytes) to maintain backwards compatibility if no units are provided
     this.fileBufferSizeBytes = (int) conf.getSizeAsKb("spark.shuffle.file.buffer", "32k") * 1024;
     this.numElementsForSpillThreshold =
       conf.getLong("spark.shuffle.spill.numElementsForceSpillThreshold", Long.MAX_VALUE);
-    this.pageSizeBytes = (int) Math.min(
-      PackedRecordPointer.MAXIMUM_PAGE_SIZE_BYTES, taskMemoryManager.pageSizeBytes());
```

Comment:
Is it safe to make this change? After this change, `PackedRecordPointer.MAXIMUM_PAGE_SIZE_BYTES` is only used in test code. The packed record pointer is used to compress pointers to addresses inside of memory pages (for use in the optimized sorter here) and thus imposes a limit on the size of pages that contain multiple sorted records. If we omit this check then we might run into corruption or other issues if `taskMemoryManager.pageSizeBytes()` is bigger than `PackedRecordPointer.MAXIMUM_PAGE_SIZE_BYTES`. This is only likely to be an issue on machines with huge heap sizes, but I still think we should guard against it.